### PR TITLE
Add "why Cadence" section to Cadence docs

### PR DESCRIPTION
**Pull request by j1010001**
## Description

Adding section to the docs on why should devs use Cadence. The 4th top-level tile is added in https://github.com/onflow/cadence/pull/2040.

______

- [x] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by j1010001** [comment]
blocking https://github.com/onflow/cadence/pull/2040

File: docs/why-cadence.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+## Why should you use Cadence for building smart contracts ?
```
Comment (10thfloor) [inline docs/why-cadence.md:1]:
```suggestion
---
title: Why Use Cadence?
---

## Why should you use Cadence for building smart contracts ?
```

**Review by 10thfloor — APPROVED** [review]

File: docs/why-cadence.md (hunk 1 above)
Comment (j1010001) [reply docs/why-cadence.md:1]: I'd prefer to use the more descriptive title, the reasons are revolving around the benefits specifically for smart-contract development.

File: docs/why-cadence.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+### 1) **Security and Safety**
+
+Cadence provides security and safety guarantees that greatly simplify the development of secure smart contracts. As smart contracts often deal with valuable assets, Cadence provides the resource-oriented programming paradigm, which guarantees that assets can only exist in one location at a time, cannot be copied, and cannot be accidentally lost or deleted. 
+Cadence includes several language features which prevent entire classes of bugs.
+These security and safety features allow smart contract developers to focus on the business logic of their contract instead of preventing accidents and attacks.
+
+### 2) Composability
+
+Cadence enables composability. Resources (which are arbitrary user-defined data types) are stored directly in users’ accounts, and can flow freely between contracts: They can be passed as arguments to functions, returned from functions, or even combined in arbitrary data structures. This makes implementing business logic easier, more natural and promotes reuse of existing logic.
```
Comment (dreamsmasher) [inline docs/why-cadence.md:11]:
```suggestion
Cadence enables composability. Resources (which are arbitrary user-defined data types) are stored directly in users’ accounts, and can flow freely between contracts: they can be passed as arguments to functions, returned from functions, or even stored in arbitrary data structures. This makes implementing business logic easier, and promotes reuse of existing logic.
```

File: docs/why-cadence.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+### 3) **Simplicity**
+
+Cadence’s syntax is inspired by popular modern general-purpose programming languages like **[Swift](https://developer.apple.com/swift/)**, **[Kotlin](https://kotlinlang.org/)**, and **[Rust](https://www.rust-lang.org/),** so developers will find the syntax and the semantics familiar.
```
Comment (dreamsmasher) [inline docs/why-cadence.md:15]:
```suggestion
Cadence’s feature set and language design is inspired by popular modern general-purpose programming languages like **[Swift](https://developer.apple.com/swift/)**, **[Kotlin](https://kotlinlang.org/)**, and **[Rust](https://www.rust-lang.org/),** so developers will find the syntax and semantics familiar. 
```

**Review by dreamsmasher — APPROVED** [review]

**Review by SupunS — APPROVED** [review]

**Pull request merged**